setup-environment-internal

```diff
@@ -247,11 +247,20 @@ ln -sf "${MANIFESTS}" "${OEROOT}"/layers/
 
 DISTRO_DIRNAME=$(echo "${DISTRO}" | sed 's#[.-]#_#g')
 
+REMOTE=$(date +%s)
+git --git-dir ${MANIFESTS}/.git remote add -f --tags $REMOTE https://github.com/foundriesio/lmp-manifest &>/dev/null
+LMP_VER=$(git --git-dir ${MANIFESTS}/.git describe $REMOTE/main --tags --abbrev=0)
```

Comment:
I still don't get why we look at the foundries' remote tags, not local?
Why not just 
```
git --git-dir ${MANIFESTS}/.git describe HEAD --tags --abbrev=0
```
?
Foundries' `main` goes by its way. Local manifest - its own one. Assume, we release v90, v91, v92. And I need to fix something in v90. Or a customer still uses v90 and needs to rebuild its factory locally.
But with that code, we've got LMP_VER=92, whereas a local manifest is v90. So I presume cache won't work for such cases.

Reply:
> I still don't get why we look at the foundries' remote tags, not local?

The main reason is because this need to run as well in the costumer factories that can have other tags.

But you are right  and this is not needed because this is already handled by the ci-script setup, I will remove this remote and back again to the initial version of the PR that get the tags from the local HEAD

Reply:
> I still don't get why we look at the foundries' remote tags, not local?

the main reason is that you can manage your version of lmp-manifest without pulling in tags from foundriesio/lmp-manifest. some factories do this. `git fetch upstream` doesn't get tags, so maintainers have to remember to also run `git fetch upstream --tags`. But then jose's reason also comes into play.

Reply:
I think we can go with what we have now that is not perfect but will be acceptable most of the time.

One bug that I see is when we have a tag `ZZZ` between the `HEAD` and the `LMP_BASE_COMMIT` or the release tag `vXX`, for such cases `LMP_VER` will be equal to this `ZZZ` tag.

Reply:
Sure, we can improve it later :)